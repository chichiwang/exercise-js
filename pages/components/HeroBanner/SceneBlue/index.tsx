import React from 'react';
import classNames from 'classnames';

import styles from './styles.module.scss';

function SceneBlue() {
  return (
    <div className={styles.field}>
      <div className={classNames(styles.hex, styles.ripple)} />
      <div className={classNames(styles.hex, styles.ripple, styles.delay1)} />
      <div className={classNames(styles.hex, styles.ripple, styles.delay2)} />
    </div>
  );
}

export default SceneBlue;
