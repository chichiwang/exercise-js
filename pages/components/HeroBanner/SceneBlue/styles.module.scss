@use "sass:map";
@use "app/var.module";

.field {
  position: relative;
  width: 100%;
  height: 138px;
  background-color: map.get(var.$colors, 'blue');
  overflow: hidden;

  .hex {
    position: absolute;
    top: 50%;
    left: 50%;
    height: 27px;
    width: 46px;
    transform: rotateX(60deg) translate(-50%, 15%);
    background-color: map.get(var.$colors, 'white');
    border-radius: 3%;
    opacity: 0;

    &::before,
    &::after {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      background-color: map.get(var.$colors, 'white');
      border-radius: 3%;
      height: 100%;
    }

    &::before {
      transform: rotate(60deg);
    }

    &::after {
      transform: rotate(-60deg);
    }
  }
}

/**
 * Keyframe animations
 */

$animation-delay: 600ms;

.ripple {
  animation-iteration-count: infinite;
  animation-duration: 9000ms;
  animation-name: ripple;
  animation-timing-function: ease-out;
  animation-delay: $animation-delay;

  &.delay1 {
    animation-delay: calc($animation-delay * 1.625);
  }

  &.delay2 {
    animation-delay: calc($animation-delay * 2.125);
  }

  @media print {
    animation: none;
  }
}

@keyframes ripple {
  0% {
    height: 27px;
    width: 46px;
    transform: rotateX(60deg) translate(-50%, 15%);
    opacity: 0;
  }

  90% {
    height: 27px;
    width: 46px;
    transform: rotateX(60deg) translate(-50%, 15%);
    opacity: 0;
  }

  94% {
    opacity: 0.15;
  }

  99% {
    opacity: 0;
  }

  100% {
    height: 108px;
    width: 184px;
    transform: rotateX(60deg) translate(-50%, -70%);
    opacity: 0;
  }
}
