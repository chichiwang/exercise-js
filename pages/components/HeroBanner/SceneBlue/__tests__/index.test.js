import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import SceneBlue from '../index';

describe('HeroBanner - SceneBlue', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<SceneBlue />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
