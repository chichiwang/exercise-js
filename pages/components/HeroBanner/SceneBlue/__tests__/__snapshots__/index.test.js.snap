// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner - SceneBlue matches snapshot 1`] = `
<div
  className="field"
>
  <div
    className="hex ripple"
  />
  <div
    className="hex ripple delay1"
  />
  <div
    className="hex ripple delay2"
  />
</div>
`;
